arangod/RestHandler/RestReplicationHandler.cpp

```diff
@@ -1746,6 +1746,57 @@ void RestReplicationHandler::handleCommandRestoreView() {
 
   LOG_TOPIC(TRACE, Logger::REPLICATION) << "restoring view: "
     << nameSlice.copyString();
+
+  if (ServerState::instance()->isCoordinator()) {
```

Comment:
wait why is it necessary to add this if else case here and not in the RestViewHandler ?
I think we want to avoid these kind of exceptions in new code

Reply:
this logic was moved from TRI_vocbase_t::createView(...)
if (ServerState::instance()->isCoordinator()) {
since TRI_vocbase_t is only responsible for the non-coordinator case